django_pg_simple_hll/migrations/0001_initial.sql

```diff
@@ -0,0 +1,145 @@
+-- The state transition function
+-- it takes the current state, and the two input values
+-- hll_agg_state is the current running state, made up of bucket keys and hashes
+--  it is an array of n_buckets + 2
+-- input is the element we are considering
+-- hll_precision is the precision we are using for the approximation
+--  it use used to calculate the number of buckets
+create or replace function
+hll_bucket(hll_agg_state int[], input anyelement, hll_precision int)
+returns int[]
+language plpgsql immutable strict parallel safe as
+$$
+declare
+    n_buckets int := POW(2, hll_precision); -- eg.. for 512 buckets, hll_precision is 9 because 2**9 -> 512
+    hashed_input int := hashtext(input::text); -- transform the input into a text and hash it
+    bucket_hash int := hashed_input & ~(1 << 31); -- keep all the significant bits, without the sign
+    bucket_key int := hashed_input & (n_buckets - 1); -- bucket the hash into one of n buckets
```

Comment:
So only the last `hll_precision` bits of `bucket_key` actually matter?

Reply:
yes:
```python
f"{(9)  & (2**3 - 1) & 0xffffffff:0b}" -> 1
````